relatedness/get_ibd.py

```diff
@@ -0,0 +1,407 @@
+            & (ht.ibd1 > grandparent_ibd1)
+            & (ht.ibd2 < grandparent_ibd2)
+        )
+    )
+    ht = ht.annotate(pair=hl.sorted([ht.i, ht.j]))
+
+    out_summary.write(
+        f"NOTE: kinship table was filtered to:\n(kin > {first_degree_pi_hat}) or kin > {grandparent_pi_hat} and IBD1 > {grandparent_ibd1} and IBD2 > {grandparent_ibd2})\n"
+    )
+    out_summary.write(f"relationships not meeting this critera were not evaluated\n\n")
+
+    return ht
+
+
+def write_functional_pedigree(
+    input_pedigree: str, vcf_samples: list, output_dir: str, output_name: str
+) -> dict:
+
+    """
+    Write a functional pedigree (pedigree with samples not in the VCF removed) and create dictionary of sample as key and seqr project as value.
+
+    :param input_pedigree: Pedigree
+    :param vcf_samples: Dictionary of samples found in the VCF
+    :return: Dictionary of project IDs for each sample
+    """
+    seqr_projects = defaultdict(str)
+
+    out_new_ped = hl.hadoop_open(
+        f"{output_dir}/{output_name}_functioning_pedigree.ped", "w"
+    )
+
+    with hl.hadoop_open(input_pedigree, "r") as infile:
+        next(infile)
+        for line in infile:
+            line = line.rstrip("\n")
+            items = line.split("\t")
+            (
+                Project_GUID,
+                Family_ID,
+                Individual_ID,
+                Paternal_ID,
+                Maternal_ID,
+                Sex,
+            ) = items[0:6]
+
+            if Individual_ID not in vcf_samples:
+                Individual_ID = "."
+            if Paternal_ID not in vcf_samples:
+                Paternal_ID = "."
+            if Maternal_ID not in vcf_samples:
+                Maternal_ID = "."
+
+            # Only output line from pedigree if the proband is not missing
+            if Individual_ID != ".":
+                seqr_projects[Individual_ID] = Project_GUID
+                out_new_ped.write(
+                    f"{Family_ID}\t{Individual_ID}\t{Paternal_ID}\t{Maternal_ID}\t{Sex}\n"
+                )
+
+    out_new_ped.close()
+
+    return seqr_projects
+
+
+def main(args):
+    output_dir = args.output_dir
+    output_name = args.output_name
+    sex_check = args.sex_check
+    mt_path = args.mt_path
+    input_pedigree = args.input_pedigree
+
+    gnomad_ld = args.gnomad_ld
+    run_ibd = args.run_ibd
+    first_degree_pi_hat = args.first_degree_pi_hat
+    grandparent_pi_hat = args.grandparent_pi_hat
+    grandparent_ibd1 = args.grandparent_ibd1
+    grandparent_ibd2 = args.grandparent_ibd2
+
+    logger.info("Reading in inputs...")
+    mt = hl.read_matrix_table(mt_path)
+    pedigree = hl.import_table(input_pedigree, impute=True)
+
+    # Infer build of the MatrixTable
+    build = get_reference_genome(mt.locus).name
+
+    logger.info("Filtering to biallelic SNVs on autosomes and performing LD pruning...")
+    mt = filter_rows_for_qc(mt, min_af=0.001, min_callrate=0.99, apply_hard_filters=False)
```

Comment:
Looks like this isn't imported from anywhere?

Reply:
oops, yeah that was the missing import at the top